csharp/extractor/Semmle.Extraction.CIL.Driver/ExtractorOptions.cs

```diff
@@ -102,11 +99,12 @@ class AssemblyList
     {
         class AssemblyNameComparer : IEqualityComparer<AssemblyName>
         {
-            bool IEqualityComparer<AssemblyName>.Equals(AssemblyName x, AssemblyName y) =>
-                x.Name == y.Name && x.Version == y.Version;
+            bool IEqualityComparer<AssemblyName>.Equals(AssemblyName? x, AssemblyName? y) =>
+                object.ReferenceEquals(x, y) ||
+                x?.Name == y?.Name && x?.Version == y?.Version;
 
             int IEqualityComparer<AssemblyName>.GetHashCode(AssemblyName obj) =>
-                obj.Name.GetHashCode() + 7 * obj.Version.GetHashCode();
+                (obj.Name, obj.Version).GetHashCode();
```

Comment:
Nice, hadn't thought of that way to combine hash codes.

Reply:
There's another new way of computing hashcodes: `HashCode.Combine()`. I'm not sure which is the preferred way.

Reply:
Perhaps the latter does not create the intermediate tuple object.